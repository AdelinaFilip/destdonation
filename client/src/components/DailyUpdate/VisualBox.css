.test p {
  position: absolute;
  padding: auto;
  width: 60%;
}

.test {
  position: relative;
}

.test img {
  width: 100%;
}
