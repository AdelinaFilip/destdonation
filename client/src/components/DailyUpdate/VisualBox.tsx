import * as React from "react";
import "./VisualBox.css";

export interface VisualBoxProps {
  imageSrc: string;
  leftOffset: number;
  topOffset: number;
}

export interface VisualBoxState {}

class VisualBox extends React.Component<VisualBoxProps, VisualBoxState> {
  state = {};

  static defaultProps = {
    leftOffset: 0,
    topOffset: 0,
  };

  textStyle = () => {
    return {
      left: this.props.leftOffset,
      top: this.props.topOffset,
    };
  };

  render() {
    return (
      <div className="test">
        <img src={this.props.imageSrc} />
        <p style={this.textStyle()}>{this.props.children}</p>
      </div>
    );
  }
}

export default VisualBox;
