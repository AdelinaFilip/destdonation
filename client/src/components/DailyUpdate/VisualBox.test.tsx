import React from "react";
import { shallow } from "enzyme";

import VisualBox from "./VisualBox";
import van from "./van.png";

describe("VisualBox", () => {
  it("should render correctly with no positioning", () => {
    const component = shallow(<VisualBox imageSrc={van} />);

    expect(component).toMatchSnapshot();
  });

  it("should render correctly with set positioning", () => {
    const component = shallow(
      <VisualBox imageSrc={van} leftOffset={20} topOffset={35} />
    );

    expect(component).toMatchSnapshot();
  });
});
