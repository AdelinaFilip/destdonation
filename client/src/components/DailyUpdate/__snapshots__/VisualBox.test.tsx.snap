// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`VisualBox should render correctly with no positioning 1`] = `ShallowWrapper {}`;

exports[`VisualBox should render correctly with set positioning 1`] = `ShallowWrapper {}`;
